File: SGA_Android/app/src/main/java/com/example/sga/data/dto/traspasos/LineaPaletDto.kt
package com.example.sga.data.dto.traspasos

import com.google.gson.annotations.SerializedName

data class LineaPaletDto(
    val id: String,
    @SerializedName("paletId")
    val idPalet: String,
    val codigoArticulo: String,
    @SerializedName("descripcionArticulo")
    val descripcion: String?,
    val lote: String?,
    val fechaCaducidad: String?,
    val cantidad: Double,
    val ubicacion: String?
)

File: SGA_Api/SGA_Api/Models/Palet/LineaPaletDto.cs
namespace SGA_Api.Models.Palet
{
	public class LineaPaletDto
	{
		public Guid Id { get; set; }
		public Guid PaletId { get; set; }
		public short CodigoEmpresa { get; set; }
		public string CodigoArticulo { get; set; } = null!;
		public string? DescripcionArticulo { get; set; }
		public decimal Cantidad { get; set; }
		public string? UnidadMedida { get; set; }
		public string? Lote { get; set; }
		public DateTime? FechaCaducidad { get; set; }
		public string CodigoAlmacen { get; set; } = null!;
		public string Ubicacion { get; set; } = null!;
		public int UsuarioId { get; set; }
		public DateTime FechaAgregado { get; set; }
		public string? Observaciones { get; set; }

		// 🔷 NUEVO: Indicadores de bloqueo por calidad
		public bool IsBloqueadoCalidad { get; set; }
		public string? MotivoBloqueoCalidad { get; set; }
		public DateTime? FechaBloqueoCalidad { get; set; }
		public string? TipoBloqueoCalidad { get; set; }
		
		// 🔷 NUEVO: TraspasoId para obtener fecha del último traspaso
		public Guid? TraspasoId { get; set; }
	}
}
